### ISSUE TITLE
Haivision SRT player "Play Pro" not capable of streamid

### ISSUE THREAD

**Issue by triple-it** [Edward-Wu/srt-live-server#23]
Hi,
The Haivision Play Pro player is not capable of doing "?streamid" in the URLs.
It would be cool to use their mobile player to connect to SLS.

Is there some known work-around for this, to get this to work anyway?

URL in Play Pro not accepted
srt://192.168.178.24:8081?streamid=live.sls.com/live/test
~~
2020-04-04 13:03:59:112 SLS ERROR: [0x7fc168808c00]CSLSSrt::libsrt_split_sid, sid='' is not as host/app/name.
2020-04-04 13:03:59:112 SLS ERROR: [0x102ade000]CSLSListener::handler, [192.168.178.23:63903], parse sid='' failed.
~~

**Comment by Edward-Wu** [comment]
thanks.
The official app store shows that there's only iphone or ipad version of Play Pro Player. I'll test it in next day.

**Comment by Edward-Wu** [comment]
you are right, the Play Pro doesn't support the streamid, I've reported the bug to SRT.
you can keep an eye on it.
https://github.com/Haivision/srt/issues/1233

**Comment by anders-d** [comment]
Hi. VLC as player doesn't seem to pass the streamid option either.
 I solved it for now with a hardcode of the sid char in SLSListener.cpp, if sid strlen == 0.
This only works of course, if the publisher passes streamid correctly, and you are only distributing a single stream.
 A more nice way to do it would be to add a conf setting like "single-stream TRUE/FALSE (default false)
 And in case of TRUE, SLS disregards of any streamid, but also allows only a single publisher stream.

**Comment by jy02505147** [comment]
@triple-it You can use "Larix Player for moblie",or "Vmix for win".They support streamid.

**Comment by wonder-wanderer** [comment]
Regarding StreamID and VLC please see #52

**Comment by KiaanChetty** [comment]
@anders-d can you please explain where you added the static entry I have been trying to recompile it to have this exclusion but it keeps failing to compile...

**Comment by anders-d** [comment]
Hi @KiaanChetty ,

I'm not using that sid hardcode anymore, since almost all SRT players support stream ids now, and unfortunately I don't have my old source tweaks anymore.
But after looking at the source now, I think I did something like this In SLSListener.cpp:

At line 381,before the sls_log() entry, add something like:

if (NULL != sid ){
  if (strlen(sid) == 0) {
    strcpy (sid, "live.teststream.com/live/test");
    cout << "KiaanDEBUG: Hardcoding sid: " << sid;
  }
}

**Comment by blizzardster** [comment]
> @triple-it您可以使用“Larix Player for moblie”或“Vmix for win”。它们支持streamid。

Thanks for the recommended APP
